### ISSUE TITLE
Consider: Error on duplicate entries in ResolveFilesToPublish

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/sdk#10850]
If multiple entries in ResolveFilesToPublish want to be copied as the same target file,
* A normal publish succeeds, causing one of the files to be in the publish directory
* PublishSingleFile fails, since the bundler checks for duplicates.

We should reconcile this behavior.
Ideally we should pick the later, since it throws an error rather than silent unpredictable behavior.
However, this can cause several customer builds to fail, because of conflicts in nuget packages (which were hitherto undetected).

Sample repro:

For example, considering the template console app with: 

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
   <RuntimeIdentifier>win-x64</RuntimeIdentifier>
  </PropertyGroup>

  <ItemGroup>
      <PackageReference Include="Microsoft.TestPlatform.CLI" Version="16.5.0"/>
  </ItemGroup>
</Project>
```
`dotnet publish` succeeds.
`dotnet publish /p:PublishSingleFile=true` fails.

If we look at the msbuild logs, we see that there are several duplicate entries trying to be bundled into app. For example, the following two DLLs want to be published as `Microsoft.TestPlatform.CommunicationUtilities.dll`.

```
.nuget\packages\microsoft.testplatform.cli\16.5.0\contentFiles\any\netcoreapp2.1\Microsoft.TestPlatform.CommunicationUtilities.dll
  CopyToPublishDirectory=PreserveNewest
  RelativePath=Microsoft.TestPlatform.CommunicationUtilities.dll
  TargetPath=Microsoft.TestPlatform.CommunicationUtilities.dll

.nuget\packages\microsoft.testplatform.testhost\16.5.0\lib\netcoreapp2.1\Microsoft.TestPlatform.CommunicationUtilities.dll
  AssetType=runtime
  CopyLocal=true
  CopyToPublishDirectory=PreserveNewest
  DestinationSubPath=Microsoft.TestPlatform.CommunicationUtilities.dll
  NuGetPackageId=Microsoft.TestPlatform.TestHost
  NuGetPackageVersion=16.5.0
  PackageName=Microsoft.TestPlatform.TestHost
  PackageVersion=16.5.0
  PathInPackage=lib/netcoreapp2.1/Microsoft.TestPlatform.CommunicationUtilities.dll
  RelativePath=Microsoft.TestPlatform.CommunicationUtilities.dll
```

**Comment by swaroop-sridhar** [comment]
CC: @wli3 @dsplaisted

**Comment by dsplaisted** [comment]
Related: https://github.com/dotnet/sdk/issues/11959

**Comment by Imperatorn** [comment]
Any update on this?

**Comment by sfoslund** [comment]
> Any update on this?

We're working on a PR here: https://github.com/dotnet/sdk/pull/14020

**Comment by sfoslund** [comment]
Fixed by #14020
